src/test/java/com/epam/tc/hw2/exercise1/SiteStructureTests.java

```diff
@@ -0,0 +1,242 @@
+        List<WebElement> imagesOnIndexPage =
+            webDriver.findElements(By.xpath("//span[contains(@class,'icons-benefit')]"));
+        softAssertions.assertThat(imagesOnIndexPage.size())
+                      .as(" There are should be 4 images on the Index Page")
+                      .isEqualTo(4);
+
+        for (WebElement imageElement : imagesOnIndexPage) {
+            softAssertions.assertThat(imageElement.isDisplayed())
+                          .as("The image element with class = '"
+                              + imageElement.getAttribute("class")
+                              + "' should be visible").isTrue();
+        }
+
+        // 7. Assert that there are 4 texts on the Index Page under icons, and they have proper text
+        List<WebElement> benefitsTxt = webDriver.findElements(By.xpath("//span[contains(@class,'benefit-txt')]"));
+        softAssertions.assertThat(benefitsTxt.size())
+                      .as(" There are should be 4 texts on the Index Page under icons")
+                      .isEqualTo(4);
+
+        WebElement textUnderPractice = webDriver
+            .findElement(By.xpath(
+                "//span[@class ='icons-benefit icon-practise']//../..//preceding-sibling::span[@class='benefit-txt']"));
+
+        softAssertions.assertThat(textUnderPractice.getText())
+                      .as("The text under element should have proper value:")
+                      .isEqualTo("To include good practices\nand ideas from successful\nEPAM project");
+
+        WebElement textUnderCustom = webDriver
+            .findElement(By.xpath(
+                "//span[@class ='icons-benefit icon-custom']//../..//preceding-sibling::span[@class='benefit-txt']"));
+
+        softAssertions.assertThat(textUnderCustom.getText())
+                      .as("The text under element should have proper value:")
+                      .isEqualTo("To be flexible and\ncustomizable");
+
+        WebElement textUnderMulti = webDriver
+            .findElement(By.xpath(
+                "//span[@class ='icons-benefit icon-multi']//../..//preceding-sibling::span[@class='benefit-txt']"));
+
+        softAssertions.assertThat(textUnderMulti.getText())
+                      .as("The text under element should have proper value:").isEqualTo("To be multiplatform");
+
+        WebElement textUnderBase = webDriver
+            .findElement(By.xpath(
+                "//span[@class ='icons-benefit icon-base']//../..//preceding-sibling::span[@class='benefit-txt']"));
+
+        softAssertions.assertThat(textUnderBase.getText())
+                      .as("The text under element should have proper value:")
+                      .isEqualTo("Already have good base\n(about 20 internal and"
+                          + "\nsome external projects),\nwish to get more…");
+
+        // 8. Assert that there is the iframe with “Frame Button” exist
+        boolean iframeFrameExists = webDriver.findElements(By.id("frame")).size() != 0;
```

Comment:
for collections please use `! .isEmpty()` instead of `.size() != 0`
And in this specific case use appropriate matcher (e.g. nonEmpty, or not().hasSize(0))